### DISCUSSION TITLE
Garbage collection

### DISCUSSION THREAD

**Discussion by zoharl3** [yalmip/YALMIP#1038] [Possible bug or missing functionality] [answered]
I'm getting the constraints from some function that may internally define some sdpvar:

`con = constr(...);`

The problem is infeasible.

To debug it, I returned one of the intermediate expressions from the function:

`[con, interm] = constr(...);`

I do nothing with it, and yet the problem become feasible.

It's not supposed to happen (e.g. related to garbage collection), and it's just some unrelated weird thing that I'm missing, right?

**Comment by johanlofberg** [comment, marked as answer]
You would have to supply a reproducible example. Don't see at all what you mean with garbage collection (so that is not the cause)

**Reply by zoharl3** [reply]
It happened last night. It's again one of these things that I'm in the middle of something, and I just quickly patch something that bypasses the issue, and a day later I only have a vague memory of the specific example.

I'll chuck it under one of my weird late night things until it happens again.

(By garbage collection, I meant that somehow returning the expression keeps something alive where otherwise, matlab's garbage collection kills it and messes up the problem.)

Thanks for the confirmation, I'll leave it for now.
